cpp/CMakeLists.txt

```diff
@@ -79,10 +79,21 @@ option(BUILD_CUGRAPH_MG_TESTS "Build cuGraph multigpu algorithm tests" OFF)
 option(CMAKE_CUDA_LINEINFO "Enable the -lineinfo option for nvcc (useful for cuda-memcheck / profiler" OFF)
 option(BUILD_TESTS "Configure CMake to build tests" ON)
 option(USE_CUGRAPH_OPS "Enable all functions that call cugraph-ops" ON)
+option(USE_RAFT_STATIC "Build raft as a static library" OFF)
+option(CUDA_STATIC_RUNTIME "Statically link the CUDA toolkit runtime and libraries" OFF)
+option(CUGRAPH_USE_CUGRAPH_OPS_STATIC "Build and statically link the cugraph-ops library" OFF)
+option(CUGRAPH_EXCLUDE_CUGRAPH_OPS_FROM_ALL "Exclude cugraph-ops targets from cuGraph's 'all' target" OFF)
```

Comment:
How does this compare to the `USE_CUGRAPH_OPS` option above? Do we need both?

Reply:
The first option controls whether we build against cugraph-ops at all. This option just controls whether we install compiled cugraph-ops libs, which we don't want to do in wheels (mostly for size reasons).

Reply:
Ah, I didn't notice the `CUGRAPH_USE_CUGRAPH_OPS_STATIC` option, which brings the total number of cugraph-ops-related options to 3. To be clear, I was asking about the `CUGRAPH_EXCLUDE_CUGRAPH_OPS_FROM_ALL` option vs. the older `USE_CUGRAPH_OPS` option we added separately a few weeks ago ([here](https://github.com/rlratzel/cugraph/blob/872fd1c2a1d2b09ecd8d55ffefe9331c66e03be2/python/cugraph/CMakeLists.txt#L36)).

It sounds like all 3 do different things, so to answer my own question, I think we need them all.

Reply:
Yup I was answering the same one. To make sure we're on the same page:
- `USE_CUGRAPH_OPS`: Whether or not we build with cugraph-ops support. If we turn this off, some functionality in cugraph won't work.
- `CUGRAPH_USE_CUGRAPH_OPS_STATIC`: Whether to compile cugraph-ops as a static lib. We want this for wheels so that we can just compile all the necessary pieces directly into libcugraph rather than needing to ship around multiple libraries. In the more general case static linking also helps us avoid including parts of the statically linked library that are unused by the dependent library, but in this case I assume that concern is irrelevant since cugraph is the only consumer of cugraph-ops so it probably uses all of it.
- `CUGRAPH_EXCLUDE_CUGRAPH_OPS_FROM_ALL`: Turning this off prevents installation of libcugraphops.[so|a] into the install tree. We want this for wheels because once we have statically linked against it the Python package doesn't need it anymore. Since we're not making libcugraph.so publicly available to consumers (it's only there for pylibcugraph/cugraph to use) there's no reason to provide the underlying cugraph-ops library either, and removing this saves space in the wheel.

Reply:
excellent, thanks for the explanations, I may include these in a future comment to the `CMakeLists.txt` file.